### PR TITLE
Added file-readability check to analyzeBamHeade.sh.

### PR DESCRIPTION
**Pull request by vinjana**
* Improved code formatting.
 * Added better error checking on input files in some Perl files (online read tests while reading from file; better error message -- [compare here](https://perldoc.perl.org/functions/readline.html))

Turn off whitespace differences!

### CODE REVIEW

**Comment by vinjana** [comment]
To me 1.4.1 seems like the appropriate version number (only added runtime robustness code and refactored).

File: COWorkflowsBasePlugin_1.4.iml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dankwart-de) [inline COWorkflowsBasePlugin_1.4.iml]: From IDEAs side, consider to also add the patch number to the iml file:  COWorkflowsBasePlugin_1.4.1.iml
This way, IDEA can load multiple plugins more easy.

File: resources/analysisTools/tools/bamReadPairExtractor.pl [hunk 2]
```diff
@@ -123,361 +124,387 @@ BEGIN
...
-        ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-        $fields2->[10] = reverse $fields2->[10];
-        if ($flags2->{M_UNMAPPED}) {
-          ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-          $fields->[10] = reverse $fields->[10];
+        if (!$selected) {
+            delete($readHash->{$fields->[0]});
+            $hs--;
+            next;
+        }
```
Comment (dankwart-de) [inline resources/analysisTools/tools/bamReadPairExtractor.pl]: ?

File: resources/analysisTools/tools/bamReadPairExtractor.pl [hunk 3]
```diff
@@ -123,361 +124,387 @@ BEGIN
...
+                }
+                if (SCARF) {
+                    printf {$sc{u}} "%s/%s\:%s\:%s\n", $fields2->[0] =~ s/[^\d:]//rg, $matenr2, $fields2->[9], $fields2->[10] =~ tr/[\x{21}-\x{5F}]/[\x{40}-\x{7E}]/r;
+                }
+                COUNT && $count{single_read_deleted}++;
+                delete($readHash->{$fields->[0]});
+                $hs--;
+                next;
+            }
+            elsif (MINREADLENGTH() > 0 && $seqlength2 < MINREADLENGTH() && $seqlength >= MINREADLENGTH()) {
```
Comment (dankwart-de) [inline resources/analysisTools/tools/bamReadPairExtractor.pl]: Why do you place elsif on a new line?

File: resources/analysisTools/tools/bamReadPairExtractor.pl (hunk 2 above)
Comment (vinjana) [reply resources/analysisTools/tools/bamReadPairExtractor.pl]: Hm, seems like the auto code cleanup in Idea is incorrectly configured for Perl.

File: resources/analysisTools/tools/bamReadPairExtractor.pl (hunk 3 above)
Comment (vinjana) [reply resources/analysisTools/tools/bamReadPairExtractor.pl]: I overlooked that this was caused by the auto-code-cleanup function in Idea.

**Pull request merged**